baseplate/clients/sqlalchemy.py

```diff
@@ -45,11 +45,17 @@ def engine_from_config(
         credentials from ``secrets`` as a :py:class:`~baseplate.lib.secrets.CredentialSecret`.
         If this is supplied, any credentials given in ``url`` we be replaced by
         these.
+    * ``pool_recycle`` (optional): this setting causes the pool to recycle connections after
+        the given number of seconds has passed. It defaults to -1, or no timeout.
 
     """
     assert prefix.endswith(".")
     parser = config.SpecParser(
-        {"url": config.String, "credentials_secret": config.Optional(config.String)}
+        {
+            "url": config.String,
+            "credentials_secret": config.Optional(config.String),
+            "pool_recycle": config.Optional(config.String),
```

Comment:
since this is an integer, you should use `config.Optional(config.Integer)`. then the config parser will `int()` it for you and yell during config parsing (rather than later on) if the user has an invalid value in config.

Reply:
yes. I tried it but i was getting an error that said something like 
```
int() can't convert non-string with explicit base
```

Then I noticed that growth ran into something similar as well as switched to using String
https://reddit.slack.com/archives/CBDQ2E41X/p1540326246000100

Reply:
That's because you've got an integer in your test. If you make that a string it'll be like reading from a real config and all will go fine.